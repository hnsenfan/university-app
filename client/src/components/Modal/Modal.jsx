import PropTypes from 'prop-types'

import Close from '../Icons/General/Close'

import './Modal.scss'

const Modal = ({ handleClose, show, children }) => {
  const showHideClassName = show ? "modal display-block" : "modal display-none"
  return (
    <div className={showHideClassName}>
      <section className="modal-main">
        <div className='close-btn' onClick={handleClose}>
          <Close />
        </div>
        {children}
      </section>
    </div>
  )
}

Modal.propTypes = {
  show: PropTypes.bool,
  children: PropTypes.array,
  handleClose: PropTypes.func
}

export default Modal