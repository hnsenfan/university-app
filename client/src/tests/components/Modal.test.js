import { shallow } from 'enzyme'
import Modal from '../../components/Modal/Modal'

test('should test Modal component', () => {
 const wrapper = shallow(<Modal />)
 expect(wrapper).toMatchSnapshot()
})

test('should handle handleClose handler', () => {
  const onSubmitMock = jest.fn()
  const wrapper = shallow(<Modal handleClose={onSubmitMock} />)
  expect(wrapper).toMatchSnapshot()
  wrapper.find('.close-btn').simulate('click')
  expect(onSubmitMock).toHaveBeenCalled()
 })