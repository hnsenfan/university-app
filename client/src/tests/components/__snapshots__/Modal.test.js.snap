// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle handleClose handler 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main"
  >
    <div
      className="close-btn"
      onClick={[MockFunction]}
    >
      <Close />
    </div>
  </section>
</div>
`;

exports[`should test Modal component 1`] = `
<div
  className="modal display-none"
>
  <section
    className="modal-main"
  >
    <div
      className="close-btn"
    >
      <Close />
    </div>
  </section>
</div>
`;
